gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineJavaVersionIntegrationTest.groovy

```diff
@@ -341,6 +344,86 @@ class BaselineJavaVersionIntegrationTest extends IntegrationSpec {
         gradleVersionNumber << GRADLE_TEST_VERSIONS
     }
 
+    def '#gradleVersionNumber: when setupJdkToolchains=true toolchains are configured by jdks-latest'() {
+        when:
+        // language=gradle
+        buildFile << '''
+        apply plugin: 'com.palantir.jdks.latest'
+
+        javaVersions {
+            libraryTarget = 11
+            runtime = 21
+            setupJdkToolchains = true
+        }
+        java {
+            toolchain {
+                languageVersion = JavaLanguageVersion.of(11)
+                vendor = JvmVendorSpec.ADOPTIUM
+            }
+            toolchain {
+                languageVersion = JavaLanguageVersion.of(21)
+                vendor = JvmVendorSpec.ADOPTIUM
+            }
+        }
+        '''.stripIndent(true)
+        file('src/main/java/Main.java') << java11CompatibleCode
+        File compiledClass = new File(projectDir, "build/classes/java/main/Main.class")
+
+        then:
+        ExecutionResult result = runTasksSuccessfully('compileJava', 'run')
+        extractCompileToolchain(result.standardOutput).contains("gradle-jdks/amazon-corretto-11.")
+        extractRunJavaCommand(result.standardOutput).contains("gradle-jdks/amazon-corretto-21.")
+        assertBytecodeVersion(compiledClass, JAVA_11_BYTECODE, NOT_ENABLE_PREVIEW_BYTECODE)
+
+        then:
+        runTasksSuccessfully('compileJava', 'run', '-Porg.gradle.java.installations.auto-detect=false', '-Porg.gradle.java.installations.auto-download=false')
+
+        where:
+        gradleVersionNumber << GRADLE_TEST_VERSIONS
+    }
+
+    def '#gradleVersionNumber: when setupJdkToolchains=false no toolchains are configured by gradle-baseline'() {
+        when:
+        // language=gradle
+        buildFile << '''
+        apply plugin: 'com.palantir.jdks.latest'
+
+        javaVersions {
+            libraryTarget = 11
+            runtime = 21
+            setupJdkToolchains = false
+        }
+        java {
+            toolchain {
+                languageVersion = JavaLanguageVersion.of(11)
+                vendor = JvmVendorSpec.ADOPTIUM
+            }
+            toolchain {
+                languageVersion = JavaLanguageVersion.of(21)
+                vendor = JvmVendorSpec.ADOPTIUM
+            }
+        }
+        '''.stripIndent(true)
+        file('src/main/java/Main.java') << java11CompatibleCode
+        File compiledClass = new File(projectDir, "build/classes/java/main/Main.class")
+
+        then:
+        ExecutionResult result = runTasksSuccessfully('compileJava', 'run')
```

Comment:
I normally put this in `when:` and all the setup stuff before `when:`, just make this a bit more readable/immediately see what this is running.